sfepy/solvers/ls.py

```diff
@@ -805,16 +805,24 @@ class MUMPSSolver(LinearSolver):
     @staticmethod
     def coo_is_symmetric(mtx, tol=1e-9):
         """Symmetry check of the sparse matrix."""
-        a_at = mtx - mtx.T
+        row, col, data = mtx.row, mtx.col, mtx.data
 
-        if a_at.nnz == 0 or nm.all(nm.abs(a_at.data) < tol):
-            return True
+        out_of_diag = (row != col)
+        row, col, data = row[out_of_diag], col[out_of_diag], data[out_of_diag]
 
-        norm = nm.linalg.norm(mtx.data)
-        if nm.all(nm.abs(a_at.data) < tol * norm):
-            return True
+        idxs_u = nm.where(col > row)[0]
+        idxs_l = nm.where(col < row)[0]
+        if idxs_l.shape[0] != idxs_u.shape[0]:
+            return False
 
-        return False
+        iu = nm.lexsort((row[idxs_u], col[idxs_u]))
+        il = nm.lexsort((col[idxs_l], row[idxs_l]))
+
+        err = nm.abs(data[idxs_u[iu]] - data[idxs_l[il]])
+        if not nm.all(err/nm.abs(data[idxs_u].max()) < tol):
```

Comment:
Maybe `nm.any()` would be faster as it can short-citcuit.
```
if nm.any(err/nm.abs(data[idxs_u].max()) >= tol):
```

Reply:
You are right. Updated.